### PR TITLE
pacman: 5.2.2 -> 6.0.0

### PR DESCRIPTION
**Pull request by felixsinger**
###### Motivation for this change
Update to latest stable release. Also, apply the following patches
fixing various bugs.

https://gitlab.archlinux.org/pacman/pacman/-/commit/3401f9e142ac4c701cd98c52618cb13164f2146b.patch
https://gitlab.archlinux.org/pacman/pacman/-/commit/542910d684191eb7f25ddc5d3d8fe3060028a267.patch

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fortuneteller2k** [comment]
@ofborg eval

**Comment by cole-h** [comment]
@ofborg eval

(Sorry, ofborg's certs recently had an issue, meaning that it was unable to
react to PRs. This has been fixed, but all PRs filed in this period will need
ofborg to be manually kicked off.)

File: pkgs/tools/package-management/pacman/default.nix [hunk 1]
```diff
@@ -1,30 +1,29 @@
...
   src = fetchurl {
-    url = "https://sources.archlinux.org/other/${pname}/${pname}-${version}.tar.gz";
-    sha256 = "1829jcc300fxidr3cahx5kpnxkpg500daqgn2782hg5m5ygil85v";
+    url = "https://sources.archlinux.org/other/${pname}/${pname}-${version}.tar.xz";
+    sha256 = "0228bn8gk25hk816x80i064nnv2mnwgmvpd4l33vsivpb844hi00";
   };
 
-  enableParallelBuilding = true;
+  nativeBuildInputs = [ pkg-config meson ninja python3 ];
+  buildInputs = [ asciidoc gpgme curl perl libarchive openssl zlib bzip2 xz ];
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/pacman/default.nix:14]:
```suggestion
  nativeBuildInputs = [ asciidoc pkg-config meson ninja python3 ];
  buildInputs = [ gpgme curl perl libarchive openssl zlib bzip2 xz ];
```

File: pkgs/tools/package-management/pacman/default.nix [hunk 2]
```diff
@@ -1,30 +1,38 @@
-{ stdenv, lib, fetchurl, pkg-config, m4, perl, libarchive, openssl, zlib, bzip2,
-xz, curl, runtimeShell }:
+{ fetchpatch, fetchurl, lib, meson, ninja, stdenv, pkg-config, python3,
+  asciidoc, bzip2, curl, gpgme, libarchive, openssl, runtimeShell, xz, zlib }:
```
Comment (piegamesde) [inline pkgs/tools/package-management/pacman/default.nix:2]: While you're at it, can you please break it down into multiple lines?

**Comment by piegamesde** [comment]
Tried it out, doesn't work:

```
error: failed to initialize alpm library:
(root: /, dbpath: /nix/store/8klyg28y8hss4cknb0wdpyhy523hsz3k-pacman-6.0.0/var/lib/pacman/)
could not create database
```

seems like it wants to write something into the Nix store because some path is broken.

**Comment by felixsinger** [comment]
Superseded by #161115

**Pull request closed without merging**